janusgraph-core/src/main/java/org/janusgraph/diskstorage/indexing/IndexFeatures.java

```diff
@@ -43,7 +43,7 @@ public IndexFeatures(boolean supportsDocumentTTL, Mapping defaultMap, ImmutableS
                          String wildcardField, ImmutableSet<Cardinality> supportedCardinaities, boolean supportsNanoseconds,
                          boolean supportCustomAnalyzer, boolean supportsGeoContains) {
 
-        Preconditions.checkArgument(defaultMap!=null || defaultMap!=Mapping.DEFAULT);
```

Comment:
Why are we dropping the second check here? Was this meant to be `&&` instead of `||`, perhaps?